### ISSUE TITLE
Have `static_file` set the content type

### ISSUE THREAD

**Issue by defunkt** [toastdriven/itty#5]
It'd be nice if `static_file` would try to guess the content type for us or allow us to set it explicitly. Here's a patch which makes it so:

http://github.com/defunkt/itty/commit/static_file_sets_content_type

**Comment by defnull** [comment]
Guessing a mimetype is actually very easy.
(copy&paste from Bottle send_file)

```
import mimetypes
def send_file(filename, ..., guessmime = True, mimetype = None):
    ....
    if guessmime and not mimetype:
        mimetype = mimetypes.guess_type(filename)[0]
    if not mimetype: mimetype = 'text/plain'
    ....
```

But be careful: The python mimetypes module is not completely thread save. It must be initialized in the main thread. # Python Issue #5853

**Comment by defunkt** [comment]
itty already includes guessing content_type: http://github.com/toastdriven/itty/blob/master/itty.py#L262-275

My patch merely calls that method for you in `static_file`.

**Comment by toastdriven** [comment]
Again, the link doesn't seem to work for me. Help?

**Comment by defunkt** [comment]
Link should work now, sorry about that!

**Comment by toastdriven** [comment]
Fixed in SHA: dde698a7e6a0fc9c16787d91a0142ddc0e4177a1
